### ISSUE TITLE
RE:Signal for iOS/iPadOS is taking up too much space even though all media is deleted

### ISSUE THREAD

**Issue by desmond132518** [signalapp/Signal-iOS#5767]
I am a fan (Not sure if I will still be in the future). This is so disappointing that a critical issue was reported and no solid solution was delivered by the dev team. What we got was being ignored and ignored. Many unuseful features were delivered. I had lots of updates but still couldn't get this issue fixed.

I used Signal for several years, the storage space took up to 15GB+ of my space, which is far more than WhatsApp or any other instant messaging app. I couldn't find the large-size media files from the storage section on WhatsApp. Even though I manually removed the message of the huge file, I didn't retain my storage back. It's been a long while since 2021.

Please look into the issue below @charlesmchen-signal @michaelkirk-signal @michelle-signal @max-signal @harry-signal 
#4916 Signal for iOS/iPadOS is taking up too much space even though all media is deleted

**Comment by joshaustintech** [comment]
I have ended donations to Signal over #4916 being ignored. I really hope this gets addressed and that this app is not treated as Open Source In Name Only.

**Comment by desmond132518** [comment]
@joshaustintech same here

**Comment by dfberger** [comment]
I too have discontinued donations, and am nearing the point where I'm forced to actively encourage correspondence via WhatsApp. The inability to manage space usage is a critical bug, and it going unresolved while shallow cosmetic changes are prioritized aband shipped is a bit baffling.

**Comment by supersophie** [comment]
The app size keeps growing. I do not know how to stop it.  At what point shall I quit Signal, and move back to WhatsApp for work, and Matrix servers for personal comms?

**Comment by supersophie** [comment]
No progress?

**Comment by desmond132518** [comment]
What I would do is to delete the app directly when it gets over 10 Gb. Meanwhile I have the desktop version to keep the full chat history. Since they are not doing a lot of things. I will have to figure it out myself.

**Comment by harry-signal** [comment]
Hi all, thank you for your reports! We have made progress in this space, please see the discussion here: https://github.com/signalapp/Signal-iOS/issues/4916#issuecomment-2398529071

**Comment by supersophie** [comment]
The comment is broken so we do not know what progress was made.
![Uploading IMG_2357.jpeg…]()


https://github.com/signalapp/Signal-iOS/issues/4916#issuecomment-2398529071

**Comment by harry-signal** [comment]
Hm that's odd. The issue is https://github.com/signalapp/Signal-iOS/issues/4916, that link should most definitely work.

**Comment by ronnicek** [comment]
Could there be option to keep Signal size in Settings to some value as it’s in Telegram (eg.  max 2, 4, 10GB)? It would delete automatically stuff from phone and you can redownload it if needed. 

I switched my family to Signal and they are quite unhappy due to size which Signal is taking in their phone.

**Comment by dfberger** [comment]
I've moved chats back to WhatsApp for exactly this reason. It's rather frustrating.

**Comment by hwschuur** [comment]
> I've moved chats back to WhatsApp for exactly this reason

same here. months in I discovered storage management was (is) basically non-existent on iOS. In my opinion (structural) Signal usage would increase even more with support for proper storage and media management.

> they're not going to fix it. I gave up months ago.

sounds like this feature isn't planned for development or published on a roadmap? does anyone know what is the limiting/delaying factor here?

edit: just went through [the original thread](https://github.com/signalapp/Signal-iOS/issues/4916) and found [this explanation](https://github.com/signalapp/Signal-iOS/issues/4916#issuecomment-1977219224) and [the first](https://github.com/signalapp/Signal-iOS/issues/4916#issuecomment-2398529071) and [second](https://github.com/signalapp/Signal-iOS/issues/4916#issuecomment-2555966791) follow-up from @harry-signal 🙏

**Comment by mocolicious** [comment]
> same here. months in I discovered storage management was (is) basically non-existent on iOS. In my opinion (structural) Signal usage would increase even more with support for proper storage and media management.

It should be more convenient to clear your device without deleting the chat history on all of your devices too.  if there was like a delete local or something that would be ideal

**Comment by wrinkledeth** [comment]
This is a big issue affecting the future of Signal. Please listen to us devs

**Comment by supersophie** [comment]
> This is a big issue affecting the future of Signal. Please listen to us devs

It has been at least 1.3 years since this bug was lodged. The devs are not listening.  

In my opinion this is a security issue. I will not enter the US, or other oppressive regimes, with Signal installed on my main phone - The data is not fully deleted when I delete it which makes it recoverable by some agent with the means.

My family and friends have mostly started to use Matrix clients e.g Element X, or Fluffy Chat.  Although we are a minority for now.

**Comment by hwschuur** [comment]
Adding some activity to prevent closing

**Comment by supersophie** [comment]
It is stale because Signal have not done much about it such a glaring security problem - I want comprising data to be deleted when I go through Hong Kong immigration.

**Comment by cdraeger** [comment]
I also came here after investigation and repeated searches about the total lack of meaningful data and media storage management. It's baffling to me how users are supposed to manage their storage on iOS/iPadOS. At this point (where probably many users are after a while) I really need sophisticated control, global or fine-grained message or media-only retention time, and global batch deletion based on time, size, media type etc. Otherwise I have to delete all data, or the app and switch back to WhatsApp. Which I don't want.

As I've understood comments in the closed issue, foundational rework was done, but why does it stop here? We are by far not there, and the feature gap to Android is so obvious for years just by looking at the official documentation.

**Comment by supersophie** [comment]
> I also came here after investigation and repeated searches about the total lack of meaningful data and media storage management. It's baffling to me how users are supposed to manage their storage on iOS/iPadOS. At this point (where probably many users are after a while) I really need sophisticated control, global or fine-grained message or media-only retention time, and global batch deletion based on time, size, media type etc. Otherwise I have to delete all data, or the app and switch back to WhatsApp. Which I don't want.
> 
> As I've understood comments in the closed issue, foundational rework was done, but why does it stop here? We are by far not there, and the feature gap to Android is so obvious for years just by looking at the official documentation.


 Signal Foundation did nothing meaningful: I discontinued my donations a while back as a form of protest.

**Comment by SwampyApple** [comment]
It's absolutely ridiculous that not even a statement can be made on this matter. It is an absolute core issue with the application and one that significantly hinders users and sustained adoption. There is no proper data management, no option to clear out media only for certain group chats, no deduplication of forwarded/duplicate media.

Now I understand some of these are very much non trivial but to close #4916 and ignore the followup specifically is baffling. If this project should purely be seen as a hobby then remove the donation elements from the application and recommend users to switch to other applications if core functionality is to be expected. It's been 6 years since the original issue has been created.

**Comment by hwschuur** [comment]
What worries me a bit is that this issue still hasn't gotten the 'acknowledged' label assigned

**Comment by hwschuur** [comment]
Adding some activity to prevent closing

**Comment by supersophie** [comment]
Dear Signal,
This issue is very much active and continues to plague your clients.

**Comment by igisho** [comment]
Still active issue!

**Comment by wlo2** [comment]
Very relevant.

**Comment by supersophie** [comment]
Still need mending. Thank-you Signal.

...

**Comment by Sutherlandon** [comment]
I'm at 15+ GB on a 128GB phone, by far the highest usage of any app on my iPhone 13 Pro.  What do I do guys?!  Just a simple everything 30 days or 1 year old get's deleted option like iMessage.  I would prefer more options, but I would even take that.  4 years of family media messages really adds up!

Edit: I deleted manually 1GB of media from a single conversation and it did free up on iOS right away, so it's not holding any storage hostage for me, but still having to play wackamole on files 4 years old is not much fun.

**Comment by Guillawme** [comment]
I agree, it would be nice to have this option to automatically delete messages after a certain time (iMessage does it well). But it should be distinct from "disappearing messages": it should delete messages only for me, not for the other people in the conversation (who might want to keep them and might have a different storage situation).

**Comment by supersophie** [comment]
> it would be nice to have this option to automatically delete messages after a certain time (iMessage does it well). 

Hi, 
Your commend is not on the subject of this bug report. We should stick to the ongoing problem with Signal. Please create a separate feature request for Signal for your "disappearing messages".  Because I would hate ti see Signal close this because the request/bug report had changed in their opinion. Let's keep the pressure up! 

The problem with Signal is the messages likelt remain after in-app deletion.  The user thinks they're deleted the messages, but in fact these are not deleted, and can be retrieved using forensic techniques. If per chance the mysteries are actually deleted (we can ) the space is not reclaimed : we may delete one gigabyte of messages in Signal, but the use of signal will still be one gigabyte.

**Comment by SwampyApple** [comment]
Checking in after another 6 months.

**Comment by supersophie** [comment]
> Checking in after another 6 months.

Good idea. Glad you had not forgotten. Unlike Signal.
